code_search/src/code_search/t2t/similarity_transformer.py

```diff
@@ -73,5 +72,9 @@ def encode(self, features, input_key):
 
   def infer(self, features=None, **kwargs):
     del kwargs
+
+    if 'targets' not in features:
```

Comment:
Why do we need to check if targets is in features? 
Is the prediction in inference mode the concatenation of code and query embeddings?
If it is then do we need this if statement because the user can control what embeddings are computed
1) setting inputs and targets to zeros or non-zero values 
2) Which elements of the output vector they look at.

Reply:
The eval step during the training also uses this function and deliberately does not send in targets. I added this to make sure they work. Certainly not ideal though.

Reply:
It is kind of the client's responsibility to slice and dice the I/O vectors. This part is mostly safeguard the eval step of T2T.

Reply:
Can you put that in a comment?

Reply:
I think this is causing problems during inference using TFServing.

When I send predictions to TFServing I'm getting b'{ "error": "You must feed a value for placeholder tensor \\\'Placeholder\\\' with dtype int64 and shape [?,?,1,1]\\n\\t [[{{node Placeholder}} = Placeholder[_output_shapes=[[?,?,1,1]], dtype=DT_INT64, shape=[?,?,1,1], _device=\\"/job:localhost/replica:0/task:0/device:CPU:0\\"]()]]" }'